examples/chef/common/stubs.cpp

```diff
@@ -347,22 +346,101 @@ void emberAfWakeOnLanClusterInitCallback(EndpointId endpoint)
-
-#ifdef MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
-    // set Parent Endpoint and Composition Type for an Endpoint
     EndpointId kRefEndpointId           = 1;
     EndpointId kColdCabinetEndpointId   = 2;
     EndpointId kFreezeCabinetEndpointId = 3;
-    SetTreeCompositionForEndpoint(kRefEndpointId);
-    SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
-    SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
-    // set TagList
-    SetTagList(kColdCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(refrigeratorTagList));
-    SetTagList(kFreezeCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(freezerTagList));
-#endif // MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
+    bool isRefrigerator;
+    bool isColdCabinet;
+    bool isFreezeCabinet;
+    if ((DataModelUtils::isEndpointOfDeviceType(kRefEndpointId, DataModelUtils::REFRIGERATOR_DEVICE_ID, isRefrigerator) ==
+         CHIP_NO_ERROR) &&
+        isRefrigerator)
+    {
+        ChipLogProgress(NotSpecified, "Refrigerator device type on EP: %d", kRefEndpointId);
+        SetTreeCompositionForEndpoint(kRefEndpointId);
+
+        // Cold Cabinet
+        if ((DataModelUtils::isEndpointOfDeviceType(kColdCabinetEndpointId,
+                                                    DataModelUtils::TEMPERATURE_CONTROLLED_CABINET_DEVICE_ID,
+                                                    isColdCabinet) == CHIP_NO_ERROR) &&
+            isColdCabinet)
+        {
+            SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
+            SetTagList(kColdCabinetEndpointId,
+                       Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(gRefrigeratorTagList));
+        }
+
+        // Freeze Cabinet
+        if ((DataModelUtils::isEndpointOfDeviceType(kFreezeCabinetEndpointId,
+                                                    DataModelUtils::TEMPERATURE_CONTROLLED_CABINET_DEVICE_ID,
+                                                    isFreezeCabinet) == CHIP_NO_ERROR) &&
+            isFreezeCabinet)
+        {
+            SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
+            SetTagList(kFreezeCabinetEndpointId,
+                       Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(gFreezerTagList));
+        }
+    }
+}
+
+/**
+ * This initializer is for the application having oven on EP1 and child endpoints -
+ * temperatureControlledCabinet on EP2, cooktop on EP3 and cooksurface on EP4.
+ */
+void OvenTemperatureControlledCabinetCooktopCookSurfaceInit()
+{
+    EndpointId kOvenEpId                         = 1;
+    EndpointId kTemperatureControlledCabinetEpId = 2;
+    EndpointId kCooktopEpId                      = 3;
+    EndpointId kCookSurfaceEpId                  = 4;
+    bool isOven;
+    bool isTemperatureControlledCabinet;
+    bool isCooktop;
+    bool isCookSurface;
```

Comment:
All local variables must always be default initialized. Bool is not false by default.

With the API changes I proposed above, you could just do `if (HasDeviceTypeOnEndpoint(.......))` for all the ifs, and avoid the dangling bools.